src/modsim2/similarity/metrics/pad.py

```diff
@@ -0,0 +1,322 @@
+        The kernel is not varied.
+        The trained model are appended to the object's list of models.
+
+        Args:
+            kernel_name:
+            c_values: list of C values (regularisation param) to be applied
+            gamma_values: list of gamma values (kernel coefficient) to be applied
+            degree_values: list of degree values for the poly kernel to be applied
+        """
+        for c in c_values:
+            for gamma in gamma_values:
+                for degree in degree_values:
+                    svc = SVC(kernel=kernel_name, C=c, gamma=gamma, degree=degree)
+                    svc.fit(X=self.__embed_train_data, y=self.__train_labels)
+                    self.__models.append(svc)
+
+    def __evaluate_models(self):
+        """
+        This method will evaluate the classifiers that have been built
+        The evaluation metric is the mean absolute error.
+        """
+        for svc in self.__models:
+            preds = svc.predict(self.__embed_test_data)
+            errors = np.abs(preds - self.__test_labels)
+            error = np.sum(errors) / len(preds)
+            self.__errors.append(error)
+
+    def calculate_distance(
+        self,
+        data_A: np.ndarray,
+        data_B: np.ndarray,
+        labels_A: np.ndarray,
+        labels_B: np.ndarray,
+        c_values: list,
+        kernel_name: str,
+        embedding_name: str,
+        test_proportion: float,
+        gamma_values: list = ["scale"],
+        degree_values: list = [3],
+        distinct_classes: bool = False,
+    ) -> float:
+
+        """
+        Calculates the proxy a-distance for datasets A and B.
+        As this implementation uses SVM, the attributes include
+        some parameters in sklearn's implementation of SVC. A
+        single value for the kernel name is accepted, and lists
+        of possible C values, gamma values and degree values
+        (to allow for some hyperparameter tuning).
+
+        Defaults are provided for the gamma and degree values
+        meaning these do not have to be supplied in the metrics
+        config.
+
+        Args:
+            data_A: The first image dataset
+            data_B: The second image dataset
+            labels_A: The target values for the first dataset
+            labels_B: The target values for the second dataset
+            c_values: List of C values (regularisation param) to be applied
+                      in the SVCs
+            kernel_name: The kernel to be applied in the SVCs
+            embedding_name: What feature embeddings, if any, to use for the
+                            input arrays
+            test_proportion: the proportion of the dataset to hold out for testing
+            gamma_values: list of gamma values to be applied in SVCs
+            degree_values: list of degree values to be applied in polynomial SVCs
+            distinct_classes: only used during testing - drops records in A & B to
```

Comment:
we should avoid things that are only used in testing - what's the rationale here? can this be done in the test code rather than src?

Reply:
I've removed this method and associated test completely - after talking through with Phil I think we are satisfied that the metric is behaving as we expected which was my reason for doing this in the first place. 

I do think it would be interesting to score the metrics on two datasets with distinct classes to see how they behave, but not necessary.